Minor expand/correct MacOS build instructions (#37300)
gravitational/teleport: BUILD_macos.md

  * `brew install node yarn`
  * Currently, [`yarn`](https://classic.yarnpkg.com/en/docs/install) (< 2.0.0) is required

> * Yarn uses wasm-pack, so if that's not present, we need to install it
>   ```shell
>   curl https://rustwasm.github.io/wasm-pack/installer/init.sh -sSf | sh
>   ```
> 

Reviewer: ```suggestion
* Teleport uses [wasm-pack](https://rustwasm.github.io/wasm-pack/),
  so if that's not present, we need to [install](https://rustwasm.github.io/wasm-pack/installer/) it
```

- Yarn doesn't use wasm-pack, Teleport does
- There are several ways to install it, so let's just link to the installation docs.

Author: Fair enough, thanks for the input

  * `brew install node yarn`
  * Currently, [`yarn`](https://classic.yarnpkg.com/en/docs/install) (< 2.0.0) is required

> * Teleport uses wasm-pack, so if that's not present, we need to install it. Find the wasm install docs [here](https://rustwasm.github.io/wasm-pack/installer/).

Reviewer: ```suggestion
* Teleport uses wasm-pack, so if that's not present, we need to [install it]( https://rustwasm.github.io/wasm-pack/installer/).
```

WASM and wasm-pack are not the same thing.

Author: Correct, fixed

> 

Reviewer: ```suggestion
```

Remove extra newline.

Author: Done!
